### DISCUSSION TITLE
How to add a surface pressure in SOFA?

### DISCUSSION THREAD

**Discussion by TeoREN0217** [SofaDefrost/SoftRobots#286] [Help - Using]
Hi,

I'm using SOFA to create a simulation environment with a chamber. I would like to add surface pressure to the chamber to make it deform. Is there a way to apply pressure in different shapes?

Best,
Teo

**Comment by EulalieCoevoet** [comment]
Hello Teo, 

If you haven't already, you can look at the `examples/tutorials/PneunetGripper` tutorial, in particular this [step](https://github.com/SofaDefrost/SoftRobots/blob/master/examples/tutorials/PneunetGripper/details/step6-pneumaticActuatorAndPythonScriptController.py).

If this does not answer your question, can you be a bit more specific.

Eulalie

**Reply by TeoREN0217** [reply]
Hi Eulalie,

Thank you so much for your reply! I apologize for the unclear description. Here's what I meant: 
![teo](https://github.com/SofaDefrost/SoftRobots/assets/60545892/35752ef8-cc8c-45f8-be55-cbd8f6c97800)

I noticed there is a demo about surface force, but I got confused when I ran it. I want employ some specific shape of surface force. I'm not sure if it's possible in SOFA.
Thank you for your fast reply again!

Best regards,
Tea

**Reply by EulalieCoevoet** [reply]
Okay I see. Sorry for the late reply. 

Do you know the force you want to apply or should it be part of an inverse problem:
- known force: I would use the `ConstantForceField` component with a MeshROI to select / define the points of your model you want to apply forces on, from a given geometry (mesh).
- unknown force: I would use the `ForcePointActuator` from SoftRobots.Inverse (same idea with the use of MeshROI) 

Best,
Eulalie
